pageserver/src/walreceiver.rs

```diff
@@ -207,7 +207,7 @@ fn walreceiver_main(
 
                 waldecoder.feed_bytes(data);
 
-                while let Some((lsn, recdata)) = waldecoder.poll_decode()? {
+                while let Ok(Some((lsn, recdata))) = waldecoder.poll_decode() {
```

Comment:
So we now continue processing the WAL stream, if the decoding returns an error once. That seems problematic. Firstly, AFAICS we just ignore the error; at least we should print it to the log. Secondly, what happens next? Will we retry to decode it on the next iteration, and go into infinite loop if the decoding keeps failing? Or will we skip over the WAL record and continue from the next record? Both of those alternatives seem really bad to me.

Reply:
Sorry, it was one of my first attempts to address problems with node startup in @ololobus  test.
I will revert it.